be/src/olap/rowset/segment_v2/plain_page.h

```diff
@@ -0,0 +1,170 @@
+#pragma once
+
+#include "util/coding.h"
+#include "util/faststring.h"
+#include "olap/olap_common.h"
+#include "olap/types.h"
+#include "olap/rowset/segment_v2/page_builder.h"
+#include "olap/rowset/segment_v2/page_decoder.h"
+#include "olap/rowset/segment_v2/options.h"
+
+namespace doris {
+
+namespace segment_v2 {
+
+static const size_t plainPageHeaderSize = sizeof(uint32_t) * 2;
+
+template<FieldType Type>
+class PlainPageBuilder : public PageBuilder {
+public:
+    PlainPageBuilder(const PageBuilderOptions *options) :
+            _options(options) {
+        // Reserve enough space for the block, plus a bit of slop since
+        // we often overrun the block by a few values.
+        _buffer.reserve(plainPageHeaderSize + _options->data_page_size + 1024);
+        reset();
+    }
+
+    bool is_page_full() override {
+        return _buffer.size() > _options->data_page_size;
+    }
+
+    Status add(const uint8_t *vals, size_t *count) override {
+        size_t old_size = _buffer.size();
+        _buffer.resize(old_size + *count * SIZE_OF_TYPE);
+        memcpy(&_buffer[old_size], vals, *count * SIZE_OF_TYPE);
+        _count += *count;
+        return Status::OK();
+    }
+
+    Status get_dictionary_page(Slice *dictionary_page) override {
+        return Status::NotSupported("get_dictionary_page not supported in plain page builder");
+    }
+
+    Slice finish(const rowid_t page_first_rowid) override {
+        encode_fixed32_le((uint8_t *) &_buffer[0], _count);
+        encode_fixed32_le((uint8_t *) &_buffer[4], page_first_rowid);
```

Comment:
first rowid is parsed in ColumnReader. And this page only store data, if there is null this first row id is strange here